modyn/trainer_server/internal/dataset/online_dataset.py

```diff
@@ -56,6 +58,11 @@ def __init__(
         self._key_source = SelectorKeySource(self._pipeline_id, self._trigger_id, self._selector_address)
         self._uses_weights = None
 
+        # tokenizer for NLP tasks
+        self._tokenizer = None
+        if tokenizer != "":
+            self._tokenizer = self._instantiate_tokenizer(tokenizer)
```

Comment:
I think we might be able to adjust the _transform function accordingly to include the tokenizer, right? So not only the potential torchvision transformers get composed, but if the tokenizer is there, we also include that in this function. then we don't need to check this everytime we yield a sample.

Reply:
Ok so something like this?
<img width="542" alt="image" src="https://github.com/eth-easl/modyn/assets/59933812/897e9432-6adb-4aaa-924e-5d9660885719">

Note that the tokenizer is already written to be compatible with torch compose (see [here](https://pytorch.org/tutorials/beginner/data_loading_tutorial.html))

Reply:
Yes, looks good. Although we should rename the function to something like _setup_composed_transform

Reply:
Ok done